lib/rex/proto/mssql/client.rb

```diff
@@ -116,9 +116,20 @@ def detect_platform_and_arch
           result = {}
 
           server_vars = query('select @@version')[:rows][0][0]
+          if server_vars.match?(/\b\d+\.\d+\.\d+\.\d+\s\(([^)]*)\)/)
```

Comment:
This can use capture groups similar to:
https://github.com/rapid7/metasploit-framework/blob/fd10f4d2950dacfd6e70ceaabae09ff53bc3a729/lib/postgres/postgres-pr/connection.rb#L216

Or you can pluck out the values directly

```
server_vars = query('select @@version')[:rows][0][0]
architecture = server_vars[/\b\d+\.\d+\.\d+\.\d+\s\(([^)]*)\)/, 1]
```

But named capture groups seems legit